dali/test/python/test_operator_one_hot.py

```diff
@@ -66,37 +107,85 @@ def one_hot(input):
-                         for x in range(batch_size)]
-        yield check_one_hot_operator, premade_batch, None
+    for j in range(1, 5):  # test 1..4 levels of nested 'multi-dimensional' scalars
+        layout = get_initial_layout(j)
+        for i in range(5):
+            premade_batch = random_scalar_like_tensors_batch(j)
+            yield partial(check_one_hot_operator, axis=None, expected_output_dim=1, initial_layout=layout), premade_batch
 
 
-def test_one_hot ():
+def test_one_hot():
     np.random.seed(42)
+    layout = get_initial_layout(3)
     for i in range(10):
         for axis in [-1, 0, 1, 2, 3]:
-            premade_batch = [
-                np.random.randint(
-                    0, num_classes, size=np.random.randint(2, 8, size=(3,)),
-                    dtype=np.int32) for _ in range(batch_size)]
-            yield check_one_hot_operator, premade_batch, axis
+            expected_layout = modify_layout(layout, axis)
+            premade_batch = random_3d_tensors_batch()
+            yield partial(check_one_hot_operator, initial_layout=layout), premade_batch, axis
+
+
+def test_multi_dim_one_hot_no_initial_layout():
+    np.random.seed(42)
+    for axis in [-1, 0, 1, 2, 3]:
+        premade_batch = random_3d_tensors_batch()
+        yield partial(check_one_hot_operator, axis=axis, initial_layout=""), premade_batch
+
+
+def test_one_hot_reset_layout():
+    np.random.seed(42)
+    layout = get_initial_layout(3)
+    for axis in [-1, 0, 1, 2, 3]:
+        premade_batch = random_3d_tensors_batch()
+        yield partial(check_one_hot_operator, axis=axis, initial_layout=layout, layout_axis_name=""), premade_batch
+    premade_batch = random_scalars_batch()
+    yield partial(check_one_hot_operator, initial_layout="", layout_axis_name=""), premade_batch
+    premade_batch = random_scalar_like_tensors_batch(3)
+    yield partial(check_one_hot_operator, axis=None, expected_output_dim=1, initial_layout=layout, layout_axis_name=""), premade_batch
+
+
+def test_one_hot_custom_layout_axis_name():
+    np.random.seed(42)
+    layout = get_initial_layout(3)
+    premade_batch = random_3d_tensors_batch()
+    yield partial(check_one_hot_operator, axis=-1, initial_layout=layout, layout_axis_name="X"), premade_batch
+    yield partial(check_one_hot_operator, axis=-1, initial_layout=layout, layout_axis_name=0), premade_batch
+    yield partial(check_one_hot_operator, axis=-1, initial_layout=layout, layout_axis_name=1), premade_batch
+
+
+@raises(RuntimeError)
+def test_too_long_axis_name():
+    np.random.seed(42)
+    premade_batch = random_3d_tensors_batch()
+    check_one_hot_operator(premade_batch, axis=-1,
+                           initial_layout="ABC", layout_axis_name="CD")
+
+
+@raises(RuntimeError)
+def test_lowercase_axis_name():
+    np.random.seed(42)
+    premade_batch = random_3d_tensors_batch()
+    check_one_hot_operator(premade_batch, axis=-1,
+                           initial_layout="ABC", layout_axis_name="q")
```

Comment:
```suggestion
                           initial_layout="ABC", layout_axis_name="")
```
Lowercase is OK, but I think that this should fail - we should distinguish "" from None.

Reply:
OK. So just to make sure we're on the same page - onehot by default should not add layout and only providing the param explicitly should enable the feature, is that right?

How I imagined it was that by default the 'O' is added the layout and only providing empty string disables the feature. I wasn't sure if it is possible to distinguish at the c++ operator level between user passing param=None and not providing param at all to the operator call, that's why I've chosen empty string instead for opting-out.

Reply:
done